### DISCUSSION TITLE
V4: Reset filters when switching between verticals

### DISCUSSION THREAD

**Discussion by nikosmsk** [microsoft-search/pnp-modern-search#1626] [Ideas]
**Is your feature request related to a problem? Please describe.**
When the Search Filters WP is connected to more than one V4 search results web parts (one for each vertical item),  filters applied to one vertical will also reapply when switching to another vertical. Very often possible filters in these search result sources are different and so the clicked vertical shows empty results. For example, People have no FileType, whilst documents have.

**Describe the solution you'd like**
Reset only those filters, which are not applicable to a new result source and keep the others applied.

**Describe alternatives you've considered**
There is an option to "Show selected filters" in Search Results WP, but it is less convenient and intuitive to click the icon to remove filters when you see no results after switching verticals

**Comment by bryllyg** [comment]
I'd like to second this for V3 too

**Comment by lukju** [comment]
@wobba are there any plans to fix this behavior? 
I think that the current behavior is not as expected... As @nikosmsk already mentioned, if a refiner value has been selected which isn't applicable to a result set in a different vertical, that particular result webpart shows an empty result set when its vertical link gets clicked.
best possible solution for this might be to simply reset all configured refiners as soon as the user selects a different vertical.

**Comment by lukju** [comment]
@wobba Maybe you could give me some pointers so that it is getting easier for me to implement the solution mentioned by @nikosmsk as a pull request?

**Comment by wobba** [comment]
If someone needs this and want to contribute, then feel free. I don't have to add new things these days, as I don't really need it myself, so I focus mainly on bugs which needs fixing. As for pointers you need to dig into the code and figure out how it ties together. All code based takes some getting used to. Start with the click handler for vertical switch and see what it triggers in the result web part.
